src/rsz/src/BaseMove.hh

```diff
@@ -0,0 +1,234 @@
+#include "sta/MinMax.hh"
+#include "sta/NetworkClass.hh"
+#include "sta/Path.hh"
+#include "sta/PathExpanded.hh"
+#include "sta/PortDirection.hh"
+#include "sta/Sdc.hh"
+#include "sta/StaState.hh"
+#include "sta/TimingArc.hh"
+#include "sta/TimingRole.hh"
+#include "sta/Transition.hh"
+#include "sta/UnorderedMap.hh"
+#include "utl/Logger.h"
+
+namespace rsz {
+
+using std::string;
+using std::vector;
+
+using odb::dbMaster;
+
+using odb::dbMaster;
+using odb::Point;
+
+using utl::Logger;
+
+using sta::ArcDelay;
+using sta::Cell;
+using sta::Corner;
+using sta::dbDatabase;
+using sta::dbNetwork;
+using sta::dbSta;
+using sta::DcalcAnalysisPt;
+using sta::Instance;
+using sta::InstancePinIterator;
+using sta::InstanceSet;
+using sta::LibertyCell;
+using sta::LibertyPort;
+using sta::LoadPinIndexMap;
+using sta::Net;
+using sta::NetConnectedPinIterator;
+using sta::Network;
+using sta::Path;
+using sta::PathExpanded;
+using sta::Pin;
+using sta::RiseFall;
+using sta::Slack;
+using sta::Slew;
+using sta::Vertex;
+
+using BufferedNetPtr = std::shared_ptr<BufferedNet>;
+using BufferedNetSeq = std::vector<BufferedNetPtr>;
+using InputSlews = std::array<Slew, RiseFall::index_count>;
+using TgtSlews = std::array<Slew, RiseFall::index_count>;
+
+struct SlackEstimatorParams
+{
+  Pin* driver_pin;
+  Pin* prev_driver_pin;
+  Pin* driver_input_pin;
+  Instance* driver;
+  const Path* driver_path;
+  const Path* prev_driver_path;
+  LibertyCell* driver_cell;
+  const float setup_slack_margin;
+  const Corner* corner;
+
+  SlackEstimatorParams(const float margin, const Corner* corner)
+      : setup_slack_margin(margin), corner(corner)
+  {
+    driver_pin = nullptr;
+    prev_driver_pin = nullptr;
+    driver_input_pin = nullptr;
+    driver = nullptr;
+    driver_path = nullptr;
+    prev_driver_path = nullptr;
+    driver_cell = nullptr;
+  }
+};
+
+class BaseMove : public sta::dbStaState
+{
+ public:
+  BaseMove(Resizer* resizer);
+  ~BaseMove() override = default;
+
+  virtual bool doMove(const Path* drvr_path,
+                      const int drvr_index,
+                      Slack drvr_slack,
+                      PathExpanded* expanded,
+                      float setup_slack_margin)
+  {
+    return false;
+  }
+
+  virtual const char* name() const { return "BaseMove"; }
+
+  void init();
+
+  // Analysis functions
+  // virtual double deltaSlack() = 0;
+  // virtual double deltaPower() = 0;
+  // virtual douele deltaArea() = 0;
+
+  // Accept the pending optimizations
+  void commitMoves();
+  // Abandon the pending optimizations
+  void restoreMoves();
+  // Total pending optimizations (since last checkpoint)
+  int pendingMoves() const;
+  // Whether this optimization is pending
+  int pendingMoves(Instance* inst) const;
+  // Total optimizations
+  int committedMoves() const;
+  // Whether this optimization is committed or pending
+  int countMoves(Instance* inst) const;
+  // Total accepted and pending optimizations
+  int countMoves() const;
```

Comment:
pendingMoves & committedMoves sound like they are giving you the moves rather than a count.  Perhaps numPendingMoves, numCommittedMoves, and numMoves?